cmd/clusterctl/client/move.go

```diff
@@ -65,70 +70,59 @@ type RestoreOptions struct {
 }
 
 func (c *clusterctlClient) Move(options MoveOptions) error {
-	// Get the client for interacting with the source management cluster.
-	fromCluster, err := c.clusterClientFactory(ClusterClientFactoryInput{Kubeconfig: options.FromKubeconfig})
-	if err != nil {
-		return err
-	}
+	var fromCluster, toCluster cluster.Client
+	var err error
 
-	// Ensure this command only runs against management clusters with the current Cluster API contract.
-	if err := fromCluster.ProviderInventory().CheckCAPIContract(); err != nil {
-		return err
+	// If there is a source cluster - validate and get a client
+	if options.FromKubeconfig != (Kubeconfig{}) {
+		if fromCluster, err = c.getClusterClient(options.FromKubeconfig); err != nil {
+			return err
+		}
 	}
```

Comment:
Since we are not supporting empty values for `--kubeconfig` the default kubeconfig discovery rules are not applied anymore and commands like `clusterctl move --to-kubeconfing=target-cluster.kubeconfig` dont work which was previously supported.

I think we should not break this behaviour. 

It would be nice if the following commands worked:
* `clusterctl move --to-kubeconfig=target-cluster.kubeconfig [--kubeconfig=current-cluster.kubeconfig]`
* `clusterctl move --to-folder --directory=/tmp/move-test [--kubeconfig=current-cluster.kubeconfig]`
* `clusterctl move --from-folder --directory=/tmp/move-test [--kubeconfig=current-cluster.kubeconfig]`

Reply:
With the current changes `clusterctl move --to-kubeconfing=target-cluster.kubeconfig` will fail with a nil pointer exception.

Reply:
```
$ clusterctl-dev move --to-kubeconfig second-cluster.kubeconfig
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x24e715b]

goroutine 1 [running]:
sigs.k8s.io/cluster-api/cmd/clusterctl/client.(*clusterctlClient).Move(0x0, {{{0x0, 0x0}, {0x0, 0x0}}, {{0x7ff7bfefef01, 0x19}, {0x0, 0x0}}, {0x0, ...}, ...})
        sigs.k8s.io/cluster-api/cmd/clusterctl/client/move.go:117 +0x25b
sigs.k8s.io/cluster-api/cmd/clusterctl/cmd.runMove()
        sigs.k8s.io/cluster-api/cmd/clusterctl/cmd/move.go:103 +0x218
sigs.k8s.io/cluster-api/cmd/clusterctl/cmd.glob..func15(0x3d04020, {0xc000497b00, 0x2, 0x2})
        sigs.k8s.io/cluster-api/cmd/clusterctl/cmd/move.go:53 +0x17
github.com/spf13/cobra.(*Command).execute(0x3d04020, {0xc000497ae0, 0x2, 0x2})
        github.com/spf13/cobra@v1.5.0/command.go:872 +0x624
github.com/spf13/cobra.(*Command).ExecuteC(0x3d056a0)
        github.com/spf13/cobra@v1.5.0/command.go:990 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.5.0/command.go:918
sigs.k8s.io/cluster-api/cmd/clusterctl/cmd.Execute()
        sigs.k8s.io/cluster-api/cmd/clusterctl/cmd/root.go:99 +0x25
main.main()
        sigs.k8s.io/cluster-api/cmd/clusterctl/main.go:26 +0x17
```

Reply:
Thank you for the pointers. They have been implemented.

```shell
$ ./bin/clusterctl move --to-folder --directory=/home/oscar/temp/move-test/
Performing backup...
Discovering Cluster API objects
Starting backup of Cluster API objects Clusters=1
Moving Cluster API objects ClusterClasses=1
Saving files to /home/oscar/temp/move-test/

$ ./bin/clusterctl move --from-folder --directory=/home/oscar/temp/move-test/
Performing restore...
Restoring files from /home/oscar/temp/move-test/
Restoring objects into the target cluster

$ ./bin/clusterctl move --to-kubeconfig /home/oscar/.kube/config
Performing move...
Discovering Cluster API objects
Moving Cluster API objects Clusters=1
Moving Cluster API objects ClusterClasses=1
Creating objects in the target cluster
Deleting objects from the source cluster
```